packages/web-components/src/components/ic-toggle-button-group/ic-toggle-button-group.tsx

```diff
@@ -0,0 +1,290 @@
+   */
+  @Prop() accessibleLabel?: string = "Toggle button group";
+
+  /**
+   * The appearance of the toggle button group, e.g dark, or light.
+   */
+  @Prop() appearance: IcThemeForeground = "default";
+
+  /**
+   * If `true`, the toggle button group will be set to the disabled state.
+   */
+  @Prop() disabled: boolean = false;
+
+  /**
+   * If `true`, the toggle button group will fill the width of the container.
+   */
+  @Prop() fullWidth?: boolean = false;
+
+  /**
+   * The placement of the icons in relation to the toggle button labels.
+   */
+  @Prop() iconPlacement?: "left" | "right" | "top";
+
+  /**
+   * If `true`, the toggle button group will be in loading state.
+   */
+  @Prop() loading?: boolean = false;
+
+  /**
+   * If `auto`, controls are toggled automatically when navigated to. If `manual`, the controls must be actioned to change their toggled state. The value of this prop is ignored if `selectType` is set to`multi`.
+   */
+  @Prop({ mutable: true }) selectMethod?: IcSelectMethodTypes = "manual";
+
+  /**
+   * Sets whether single or multiple options can be toggled. If `multi`, then the `selectMethod` is always `manual`.
+   */
+  @Prop() selectType?: IcSelectTypes = "single";
+
+  /**
+   * The size of the toggle buttons to be displayed. This does not affect the font size of the accessible label.
+   */
+  @Prop() size?: IcSizes = "default";
+
+  /**
+   * The variant of the toggle button.
+   */
+  @Prop({ reflect: true }) variant: "default" | "icon" = "default";
+
+  /**
+   * Emitted when a toggle button is selected.
+   */
+  @Event() icChange: EventEmitter<IcChangeEventDetail>;
+
+  @Listen("icToggleChecked")
+  selectHandler(ev: CustomEvent, tabTarget?: HTMLIcToggleButtonElement): void {
+    const allToggles = this.getAllToggleButtons();
+    let clickedToggle = ev.target as HTMLIcToggleButtonElement;
+    // tabTarget used in proxySelectHandler
+    tabTarget && tabTarget.focus();
+    if (this.selectType === "single") {
+      if (!clickedToggle) {
+        clickedToggle = tabTarget;
+      }
+      allToggles.forEach((el) => {
+        if (el.id !== clickedToggle.id && el.toggleChecked) {
+          el.toggleChecked = false;
+        }
+      });
+
+      this.icChange.emit({
+        toggleChecked: ev.detail.toggleChecked,
+        selectedOption: clickedToggle,
+      });
+    } else {
+      const toggledOptions = this.getAllToggleButtons().filter(
+        (el) => el.toggleChecked && !el.disabled
+      );
+
+      this.icChange.emit({
+        toggleChecked: toggledOptions.map((opt) => opt.toggleChecked),
+        toggledOptions: toggledOptions.map((opt) => ({
+          toggleButton: opt,
+        })),
+        selectedOption: clickedToggle,
+      });
+    }
+  }
+
+  componentWillLoad(): void {
+    this.selectType === "multi" && (this.selectMethod = "manual");
+    this.selectMethod === "auto" && this.selectType === "single";
+    document.addEventListener("keydown", (ev) => {
```

Comment:
this event listener is never removed, causing a potential memory leak. Have tested this by putting a console.log in & can see it is still logged even after the component is removed. It will need to be the exact same function instance that gets removed, so you cannot bind an inline function in the addEventListener - it will need to be a method in component so that you can do

addEventListener(this.whateverMethod)

and then

document.removeEventListener(this.whateverMethod), in the disconnectedCallback